include/api/CAnomalyJobConfig.h

```diff
@@ -201,6 +354,24 @@ class API_EXPORT CAnomalyJobConfig {
         //! The scheduled events (events apply to all detectors).
         //! Events consist of a description and a detection rule
         TStrDetectionRulePrVec m_ScheduledEvents{};
+
+        friend struct CAnomalyJobLimitTest::testAccuracy;
+        friend struct CAnomalyJobLimitTest::testLimit;
+        friend struct CAnomalyJobLimitTest::testModelledEntityCountForFixedMemoryLimit;
+        friend struct CAnomalyJobTest::testBadTimes;
+        friend struct CAnomalyJobTest::testOutOfSequence;
+        friend struct CAnomalyJobTest::testControlMessages;
+        friend struct CAnomalyJobTest::testSkipTimeControlMessage;
+        friend struct CAnomalyJobTest::testIsPersistenceNeeded;
+        friend struct CAnomalyJobTest::testModelPlot;
+        friend struct CAnomalyJobTest::testInterimResultEdgeCases;
+        friend struct CAnomalyJobTest::testRestoreFailsWithEmptyStream;
+        friend struct CForecastRunnerTest::testSummaryCount;
+        friend struct CForecastRunnerTest::testRare;
+        friend struct CForecastRunnerTest::testInsufficientData;
+        friend struct CStringStoreTest::testPersonStringPruning;
+        friend struct CStringStoreTest::testAttributeStringPruning;
+        friend struct CStringStoreTest::testInfluencerStringPruning;
```

Comment:
Hopefully the approach of a helper method in the test library will allow these to be replaced with just one friend.